plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -52,12 +48,25 @@ abstract class MavenPublishBaseExtension(
         repositoryPassword = project.providers.gradleProperty("mavenCentralPassword"),
       )
 
+    val groupId = project.provider { project.group.toString() }
+    val versionIsSnapshot = project.provider { project.versionIsSnapshot }
+    val createRepository = project.tasks.registerCreateRepository(groupId, versionIsSnapshot, buildService)
+    val stagingRepositoryId = createRepository.flatMap { it.stagingRepositoryId }
+
     project.gradlePublishing.repositories.maven { repo ->
       repo.name = "mavenCentral"
       repo.setUrl(sonatypeHost.map { it.publishingUrl(project.versionIsSnapshot, stagingRepositoryId) })
       repo.credentials(PasswordCredentials::class.java)
     }
 
+    project.gradlePublishing.publications.withType(MavenPublication::class.java).all { publication ->
+      project.afterEvaluate {
```

Comment:
Why do we need the afterEvaluate here?

Reply:
The task is only created at some point after we get the callback here when a new publication was added. Afaik there is no version of `tasks.named(...)` that will give you a callback once a task with that name is created. So it's either `afterEvaluate` or doing `tasks.configureEach` and then filtering by name and in that case I think afterEvaluate is preferable.